src/main/java/org/infinispan/tutorial/embedded/WeatherApp.java

```diff
@@ -55,6 +53,16 @@ public void fetchWeather() {
       System.out.printf("---- Fetched in %dms ----\n", System.currentTimeMillis() - start);
    }
 
+   public void computeCountryAverages() {
+      System.out.println("---- Average country temperatures ----");
+      Map<String, Double> averages = cache.entrySet().stream()
+              .collect(CacheCollectors.serializableCollector(() -> groupingBy(Map.Entry::getKey,
```

Comment:
This should have been ``e -> e.getValue().country`` instead of ``Map.Entry::getKey``.  This is already fixed in master.